index.bs

```diff
@@ -338,21 +338,31 @@ So, if the object inherits from {{EventTarget}}, add a corresponding
 [=event handler IDL attributes=] on the {{GlobalEventHandlers}} interface, instead of directly on
 the relevant element interface(s).</p>
 
+<h2 id="event-design">Event Design</h2>
+
+<h3 id="events-are-for-notification">Events are for notification</h3>
+
+Try to design DOM events to deliver after-the-fact notifications of changes. It may be tempting to try to trigger side-effects from the action of <code>dispatchEvent</code>, but in general this is <a href="https://lists.w3.org/Archives/Public/public-webapps/2014AprJun/0510.html">discouraged</a>.
+
+<h3 id="favor-asynchronous-events">Favor asynchornous events</h3>
+
+A few events in the platform are specified to dispatch synchronously. These events cause problems for engines and performance issues in applications. If you feel you may ned a synchronous event in your design, please reconsider and ask the TAG for help in redesigning the API.
```

Comment:
Do we want to go into the why these are problems? E.g., synchronous events cause re-entrancy (e.g., in a focus event handler another element is focused...) which can cause unexpected results in author code and are tricky to spec correctly. Also Sync mutation events were triggered at a time in which an operation of the DOM was partially-complete, resulting in reentrancy, breaking internal invariants, and causing platform reliability issues.